src/util/http/websocket/WebSocketNotifier.cpp

```diff
@@ -2,15 +2,34 @@
 //   Chair of Algorithms and Data Structures.
 //   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
 
-#include "WebSocketNotifier.h"
+#include "util/http/websocket/WebSocketNotifier.h"
+
+#include <boost/asio/post.hpp>
+#include <boost/asio/this_coro.hpp>
 
 namespace ad_utility::websocket {
-void WebSocketNotifier::operator()(std::string json) const {
-  webSocketManager_.addQueryStatusUpdate(owningQueryId_.toQueryId(),
-                                         std::move(json));
+
+net::awaitable<WebSocketNotifier> WebSocketNotifier::create(
+    common::OwningQueryId owningQueryId, WebSocketTracker& webSocketTracker) {
+  auto initialExecutor = co_await net::this_coro::executor;
+
+  auto distributor =
+      co_await webSocketTracker.createDistributor(owningQueryId.toQueryId());
+
+  co_await net::dispatch(initialExecutor, net::use_awaitable);
+
+  co_return WebSocketNotifier{std::move(owningQueryId), webSocketTracker,
+                              std::move(distributor),
+                              std::move(initialExecutor)};
 }
 
-WebSocketNotifier::~WebSocketNotifier() {
-  webSocketManager_.releaseQuery(owningQueryId_.toQueryId());
+void WebSocketNotifier::operator()(std::string json) const {
+  net::co_spawn(
+      executor_,
+      [distributor = distributor_,
+       json = std::move(json)]() mutable -> net::awaitable<void> {
+        co_await distributor->addQueryStatusUpdate(std::move(json));
+      },
+      net::detached);
```

Comment:
If `co_spawn` were in the `distributor` this would be undefined behaviour in the following case:
1. `distributor->addQueryStatusUpdate()` is called
2. distributor shared ptr is destructed, incrementing the count to 0 and destroying the object
3. async operation is run on a deleted object
4. pray for segfault instead of weird stuff

This is also why a copy of the shared pointer is added as closure for the lambda. The alternative would be to pass the distributor a shared pointer of itself which is arguably terrible API design